src/tabs-motion/types.js

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import * as React from 'react';
+import {
+  ORIENTATION,
+  FILL,
+  KEYBOARD_ACTIVATION,
+  STATE_CHANGE_TYPE,
+} from './constants.js';
+
+import type {OverrideT} from '../helpers/overrides.js';
+import type {IconPropsT} from '../icon/types.js';
+
+export type OrientationT = $Values<typeof ORIENTATION>;
+export type FillT = $Values<typeof FILL>;
+export type KeyboardActivationT = $Values<typeof KEYBOARD_ACTIVATION>;
+export type StateChangeT = $Values<typeof STATE_CHANGE_TYPE>;
+
+export type TabsOverridesT = {|
+  Root?: OverrideT,
+  TabList?: OverrideT,
+  TabHighlight?: OverrideT,
+  TabBorder?: OverrideT,
+|};
+
+export type TabOverridesT = {|
+  Tab?: OverrideT,
+  ArtworkContainer?: OverrideT,
+  TabPanel?: OverrideT,
+|};
+
+export type StatefulTabsStateT = {|
+  activeKey: React.Key,
+|};
+
+export type StatefulTabsActionT = {|
+  type: StateChangeT,
+  payload: React.Key,
+|};
+
+export type StatefulTabsReducerT = (
+  state: StatefulTabsStateT,
+  action: StatefulTabsActionT,
+) => StatefulTabsStateT;
+
+export type onChangeT = (params: {activeKey: React.Key}) => void;
+
+export type TabsPropsT = {|
+  children: React.Node,
+  activeKey?: React.Key,
+  disabled?: boolean,
+  fill?: FillT,
+  orientation?: OrientationT,
+  keyboardActivation?: KeyboardActivationT,
```

Comment:
Why an enum vs a boolean? Are there upcoming specs suggesting another activation option?

Reply:
good point - changed this to a bool `activateOnFocus`.